src/ContentManager.php

```diff
@@ -8,64 +8,61 @@
 
 namespace Content;
 
-use Content\Behaviour\ContentProviderInterface;
 use Content\Behaviour\PropertyHandlerInterface;
-use Symfony\Component\Filesystem\Filesystem;
-use Symfony\Component\Finder\Finder;
-use Symfony\Component\Finder\SplFileInfo;
+use Content\Provider\ContentProviderInterface;
 use Symfony\Component\PropertyAccess\PropertyAccess;
 use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
-use Symfony\Component\Serializer\SerializerInterface;
+use Symfony\Component\Serializer\Encoder\DecoderInterface;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
 
 class ContentManager
 {
-    private string $path;
-    private SerializerInterface $serializer;
-    private FileSystem $files;
+    private DecoderInterface $decoder;
+    private DenormalizerInterface $denormalizer;
     private PropertyAccessorInterface $propertyAccessor;
 
-    /** @var iterable<ContentProviderInterface> */
+    /** @var iterable<ContentProviderInterface>|ContentProviderInterface[] */
     private iterable $providers;
 
-    /** @var iterable<string, PropertyHandlerInterface> indexed by property name */
+    /** @var iterable<string, PropertyHandlerInterface>|PropertyHandlerInterface[] indexed by property name */
     private iterable $handlers;
 
-    private array $cache;
+    /** @var array<string,object> */
+    private array $cache = [];
 
     public function __construct(
-        string $path,
-        SerializerInterface $serializer,
+        DecoderInterface $decoder,
+        DenormalizerInterface $denormalizer,
         iterable $propertyHandlers,
         iterable $contentProviders,
         ?PropertyAccessorInterface $propertyAccessor = null
     ) {
-        $this->path = rtrim($path, '/');
-        $this->serializer = $serializer;
+        $this->decoder = $decoder;
+        $this->denormalizer = $denormalizer;
         $this->propertyAccessor = $propertyAccessor ?? PropertyAccess::createPropertyAccessor();
-        $this->files = new FileSystem();
         $this->providers = $contentProviders;
         $this->handlers = $propertyHandlers;
-        $this->cache = [
-            'files' => [],
-            'contents' => [],
-        ];
+
+        if (is_countable($contentProviders) && 0 === \count($contentProviders)) {
+            throw new \LogicException('At least one content provider must be configured.');
```

Comment:
Must it though?

On peut imaginer installer le bundle juste pour dumper un site static en twig. 
Pas vraiment de raison de bloquer l'utilisateur ici ?

On aura bien une exception si on essaye de fetch un contenu.

Reply:
Alright